compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2464,23 +2465,27 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         );
         let mut err = self.no_such_field_err(ident, base_ty, base.hir_id);
 
-        match *base_ty.peel_refs().kind() {
+        let has_label = match *base_ty.peel_refs().kind() {
             ty::Array(_, len) => {
                 self.maybe_suggest_array_indexing(&mut err, expr, base, ident, len);
+                false
             }
             ty::RawPtr(..) => {
                 self.suggest_first_deref_field(&mut err, expr, base, ident);
-            }
-            ty::Adt(def, _) if !def.is_enum() => {
-                self.suggest_fields_on_recordish(&mut err, expr, def, ident);
+                false
             }
             ty::Param(param_ty) => {
                 self.point_at_param_definition(&mut err, param_ty);
+                false
             }
             ty::Alias(ty::Opaque, _) => {
-                self.suggest_await_on_field_access(&mut err, ident, base, base_ty.peel_refs());
```

Comment:
nit: can you please extract this whole block to a function and not make `suggest_await_on_field_access` return a value, but instead put the functionality in `suggest_await_on_field_access`, that is responsible for the return value, in this function?

Reply:
Changed.

Reply:
I didn't mean that you should get rid of the `has_label` block, just that you either move the functionality that calculates `has_label` to its own function. As an alternative you could also just put the logic that was responsible for calculating the return value of `suggest_await_on_field_access` directly into the "`has_label` block" instead of making `suggest_await_on_field_access` have a return value.

Reply:
What do you think of the last commit instead? There's some duplication of the setting of the label, but less state to keep track of.